### PR TITLE
Move testlinter and envvarlinter from istio/istio

### PR DESCRIPTION
**Pull request by geeknoid**

### CODE REVIEW

File: cmd/envvarlinter/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# envvarlinter
+
+envvarlinter ensures that non-test files don't use os.Getenv and os.LookupEnv and instead use the functions from pkg/env.
+
+envvarlinter is based on [Checker](../README.md), and this package provides the [custom rules](rules) implementation.
```
Comment (howardjohn) [inline cmd/envvarlinter/README.md]: broken link?

File: cmd/envvarlinter/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+```go
+var Whitelist = map[string][]string{
+    "/istio/mixer/pkg/*": {"skip_issue", "short_skip"},
+    "/istio/pilot/pkg/simply_test.go": {"*"},
+}
+```
+
+## Running testlinter
+
+There are two ways to run this linter.
```
Comment (howardjohn) [inline cmd/envvarlinter/README.md]: nit - looks like one way :slightly_smiling_face:

File: cmd/testlinter/README.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+```bash
+/istio/tests/e2e/tests/simple/simple_test.go
+```
+
+### End-to-end test rules
+
+1. All skipped tests must be associated with an github issue.
+
+1. All tests should be skipped if testing.short() is true.  This makes it easier to filter out long running tests
```
Comment (howardjohn) [inline cmd/testlinter/README.md]: This comment does not seem true, our e2e tests don't do this? Maybe its just disabled. Also I realize you are just moving this over so probably best to not make any changes here in this PR if ever

File: cmd/testlinter/README.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+```plain
+/istio/mixer/simple_tests.go
+
+```
+
+### Unit Test Rules
+
+1. All skipped tests must be associated with an github issue.
+1. Must not fork a new process.
+1. Must not sleep, as unit tests are supposed to finish quickly. (Open to debate)
```
Comment (howardjohn) [inline cmd/testlinter/README.md]: this is definitely not enforced, but I wish it was

File: cmd/envvarlinter/README.md (hunk 1 above)
Comment (geeknoid) [reply cmd/envvarlinter/README.md]: Indeed. I'm surprised the markdown linter didn't catch this. Anyway, I removed the whole line.

File: cmd/envvarlinter/README.md (hunk 2 above)
Comment (geeknoid) [reply cmd/envvarlinter/README.md]: Done.

File: cmd/testlinter/README.md (hunk 4 above)
Comment (geeknoid) [reply cmd/testlinter/README.md]: Turns out several things in the readme are aspirational. I marked these things as TBD.

File: cmd/testlinter/README.md (hunk 3 above)
Comment (geeknoid) [reply cmd/testlinter/README.md]: Marked TBD

**Comment by geeknoid** [comment]
PTAL

**Comment by geeknoid** [comment]
I'm putting a hold on this PR so that I can merge it manually. For the history to be preserved properly, I need to do a rebase and merge rather than a squash.

**Pull request merged**